packages/api/src/utils/pin.js

```diff
@@ -155,15 +156,33 @@ export async function waitOkPins (cid, cluster, waitTime = MAX_PIN_STATUS_CHECK_
  */
 export async function waitAndUpdateOkPins (cid, cluster, db, waitTime = MAX_PIN_STATUS_CHECK_TIME, checkInterval = PIN_STATUS_CHECK_INTERVAL) {
   const okPins = await waitOkPins(cid, cluster, waitTime, checkInterval)
-  const pins = okPins.map((pin) => {
-    return {
-      id: pin._id,
-      status: pin.status,
-      contentCid: cid,
-      location: pin.location
-    }
-  })
-  // @ts-ignore
+  const pins = toPinsUpsert(cid, okPins)
   await db.upsertPins(pins)
   return okPins
 }
+
+/**
+ * @param {string} contentCid
+ * @param {import('@web3-storage/db/db-client-types').PinUpsertInput[]} pins
+ * @returns {import('@web3-storage/db/db-client-types').PinsUpsertInput[]}
+ */
+export function toPinsUpsert (contentCid, pins) {
+  // @ts-ignore
```

Comment:
in my defence, i have simply refactored and moved the @ts-ignore that was already there on line 166.

Reply:
ah ha. Those types are only used in this file and id is not needed by the impl.

Reply:
i have fixed